File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int n = nums.size();
        if (n == 0) return -1;
        
        int minValue = nums[0];
        for (int i = 1; i < n; i++) {
            if (nums[i] < minValue) {
                minValue = nums[i];
            }
        }
        
        return minValue;
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
          int s=0, e=nums.length-1;int min=Integer.MAX_VALUE;
        while(s<=e){
            int mid=s+(e-s)/2;
            if(nums[mid]<=min) {
                min=nums[mid];

            }

            if(nums[s]<=nums[mid]){
                if(nums[s]<=min){
                    min=nums[s];
                    
                }
                s=mid+1;
                

            }
            else{
                
                e=mid-1;
            }

        
    } return min;
    }
}